CoreUtilities/src/au/gov/asd/tac/constellation/utilities/icon/IconData.java

```diff
@@ -179,17 +188,15 @@ public int hashCode() {
 
     @Override
     public boolean equals(final Object obj) {
-        if (this == obj) {
-            return true;
-        } else if (obj == null) {
+        if (obj == null) {
             return false;
-        } else if (getClass() != obj.getClass()) {
+        }
+
+        if (!(obj instanceof IconData)) {
             return false;
-        } else {
-            // Do nothing
```

Comment:
I'm a little confused by this change. Is this meant to be equivalent?

Reply:
Its not meant to be equivalent. It first checks if the object to compare is null (yes then false), if its an instance of IconData (no then false), then finally if it wasn't null and is an instance of IconData, it then checks whether the "data" arrays match and returns true/false.

Reply:
@sol695510 I think I would keep it as it was (though some of the if conditions that only returned false could probably be merged together). I think `getClass() != obj.getClass()` is a slightly stronger condition compared to `!(obj instanceof IconData)` (the latter only implies the supertype is the same, not the actual type) and you probably don't want to spend time doing extra checks if the 2 objects are the same object (the 1st condition you removed)